components/motor/gpio/basic.go

```diff
@@ -186,39 +183,51 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 		errs = multierr.Combine(errs, m.EnablePinHigh.Set(ctx, true, extra))
 	}
 
+	m.powerPct = powerPct
+	powerPct = math.Min(powerPct, m.maxPowerPct)
+	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+
+	// pin calls on builtin baords will only take poisitve power, so the absolute value is required to be passed
+	// to pwm pins.
+	pwmPower := math.Abs(powerPct)
+
 	var pwmPin board.GPIOPin
 
 	switch m.motorType {
 	case ABPwm, DirectionPwm:
-		if math.Abs(powerPct) <= 0.001 {
+		if pwmPower <= 0.001 {
```

Comment:
hmm this also needs to be set positive too though

Reply:
The ABPWM and PWMDir dont' care about the power, but the AB switch statement is switched on powerPCT but only set using pwmPower, which is the absolute of the powerPCt being calcualted

Reply:
misread which pwm value was getting used where. my b